mobile_wallet/src/lib.rs

```diff
@@ -349,6 +325,98 @@ fn sign_message_aux(input: &str) -> anyhow::Result<String> {
     ))?)
 }
 
+#[derive(common::SerdeDeserialize)]
+struct MicroCCDAmount {
+    #[serde(rename = "microGtuAmount")]
+    amount: u64,
+}
+
+#[derive(common::SerdeDeserialize)]
+struct AccountAddressJSON {
+    address: String,
+}
+
+#[derive(common::SerdeDeserialize)]
+#[serde(untagged)]
```

Comment:
Instead of this you could do
```
#[serde(tag = "type", contents = "payload"]
```
to get the structure

```json
{
   "type": "UpdateContract",
   "payload": {
        "amount': ...,
        "address": {...},
   }
```

I would recommend using the same names as in the `TransactionType` though. That would be `Update` and `Transfer`

Reply:
I wanted to do this, but the tag we set in JS is the number, not the transaction type name.
But I guess we can just not supply the type, because the payloads are distinct?

Reply:
(I changed it to this)